hibernate-core/src/main/java/org/hibernate/engine/internal/StatefulPersistenceContext.java

```diff
@@ -108,17 +109,17 @@
 	 */
 
 	// Loaded entity instances, by EntityKey
-	private HashMap<EntityKey, Object> entitiesByKey;
+	private Map<EntityKey, Object> entitiesByKey;
```

Comment:
In previous JDK, specifying `HashMap` enjoys some performance benefit, but recent JDK eliminates the difference and we have no reason not to use interface instead for we seldom rely on HashMap specific feature in our code.